### PR TITLE
[Clean-Up] Remove Transaction pointer argument for compiled plans

### PR DESCRIPTION
**Pull request by poojanilangekar**
This PR removes the `concurrency::Transaction` pointer from the `RuntimeState` for Compiled Plans. It uses the `executor::ExecutorContext` to get the transaction pointer via the proxy function.

### CODE REVIEW

File: src/include/codegen/compilation_context.h [hunk 1]
```diff
@@ -141,18 +138,19 @@ class CompilationContext {
...
   // The mapping of an operator in the tree to its translator
   std::unordered_map<const planner::AbstractPlan *,
-                     std::unique_ptr<OperatorTranslator>> op_translators_;
+                     std::unique_ptr<OperatorTranslator>>
+      op_translators_;
 
   // The mapping of an expression somewhere in the tree to its translator
   std::unordered_map<const expression::AbstractExpression *,
-                     std::unique_ptr<ExpressionTranslator>> exp_translators_;
+                     std::unique_ptr<ExpressionTranslator>>
```
Comment (pmenon) [inline src/include/codegen/compilation_context.h]: Is this change necessary?

File: src/include/codegen/compilation_context.h [hunk 2]
```diff
@@ -141,18 +138,19 @@ class CompilationContext {
...
-  RuntimeState::StateID txn_state_id_;
   RuntimeState::StateID storage_manager_state_id_;
   RuntimeState::StateID executor_context_state_id_;
   RuntimeState::StateID query_parameters_state_id_;
 
   // The mapping of an operator in the tree to its translator
   std::unordered_map<const planner::AbstractPlan *,
-                     std::unique_ptr<OperatorTranslator>> op_translators_;
+                     std::unique_ptr<OperatorTranslator>>
+      op_translators_;
```
Comment (pmenon) [inline src/include/codegen/compilation_context.h]: Is this formatting change necessary?

File: src/include/codegen/compilation_context.h [hunk 3]
```diff
@@ -72,31 +72,28 @@ class CompilationContext {
 
   RuntimeState &GetRuntimeState() const { return query_.GetRuntimeState(); }
 
-//  const QueryParameters &GetQueryParameters() const {
-//    return parameters_;
-//  }
+  //  const QueryParameters &GetQueryParameters() const {
+  //    return parameters_;
+  //  }
```
Comment (pmenon) [inline src/include/codegen/compilation_context.h]: You can remove this commented code.

File: src/codegen/operator/table_scan_translator.cpp [hunk 4]
```diff
@@ -125,8 +127,12 @@ void TableScanTranslator::ScanConsumer::ProcessTuples(
   }
 
   // 3. Setup the (filtered) row batch and setup attribute accessors
-  RowBatch batch{translator_.GetCompilationContext(), tile_group_id_, tid_start,
-                 tid_end, selection_vector_, true};
+  RowBatch batch{translator_.GetCompilationContext(),
```
Comment (pmenon) [inline src/codegen/operator/table_scan_translator.cpp]: Was this change necessary?

File: src/codegen/compilation_context.cpp [hunk 5]
```diff
@@ -27,16 +26,14 @@ namespace codegen {
 CompilationContext::CompilationContext(Query &query,
                                        const QueryParametersMap &parameters_map,
                                        QueryResultConsumer &result_consumer)
-    : query_(query), parameters_map_(parameters_map),
+    : query_(query),
+      parameters_map_(parameters_map),
```
Comment (pmenon) [inline src/codegen/compilation_context.cpp]: Was this formatting change necessary?

**Comment by poojanilangekar** [comment]
@pmenon Sorry about the formatting error. It was because I ran the clang-format manually. I have made all changes you suggested. Let me know if you want me to make any other changes.
